src/datatypes.nim

```diff
@@ -7,22 +7,29 @@
 #
 # at your option. This file may not be copied, modified, or distributed except according to those terms.
 
-import ./private/lowlevel_types
-import ttmath
-
-export lowlevel_types, ttmath
+import ./private/conversion_bytes
 
+# Note: Fields F should be private, it is intentionally ugly to directly access them
+# See private field access issue: https://github.com/nim-lang/Nim/issues/7390
 type
   PublicKey* = object
-    raw_key*: ByteArrayBE[64]
+    Fraw_key*: array[64, byte]
 
   PrivateKey* = object
-    raw_key*: ByteArrayBE[32]
-    public_key*: PublicKey
+    Fraw_key*: array[32, byte]
+    Fpublic_key*: PublicKey
```

Comment:
Getting a public key from a private key is a perfectly valid use case, thus I'd suggest either naming this field `public_key*`, or creating a `public_key` getter, which might in future do on demand public key computation instead of storing the public key within `PrivateKey` object. The tests then should use this public api instead of `Fpublic_key`.